### PR TITLE
cabana: color bytes based on activity

### PR DESCRIPTION
**Pull request by pd0wm**
Color bytes based on their activity type. If they are constantly changing they will be colored grey. Signals that are mostly stationary are colored blue or red when they increase or decrease.

MacOS light mode:
Screenshot 2023-01-15 at 14 23 02

MacOS dark mode:
Screenshot 2023-01-15 at 14 23 14

Before:
before

### CODE REVIEW

**Comment by pd0wm** [comment]
@deanlee Let me know if there is a better place to put this than in the `eventFilter`.

**Comment by deanlee** [comment]
This is the best place right now :+1:  :D

**Comment by deanlee** [comment]
I'll add this feature to binary view and history log view.  it may only take a few lines of code based on your code!

**Comment by deanlee** [comment]
almost done.I really love this feature:)

[Kazam_screencast_00050.webm](https://user-images.githubusercontent.com/27770/212569514-43e9da5b-d2d6-4a41-8cd0-9afe87afeb7c.webm)

**Comment by adeebshihadeh** [comment]
The design and colors look great too!

**Comment by pd0wm** [comment]
@deanlee any commits you want me to merge/cherry-pick onto this branch? Or do you want to make a separate PR once this is merged?

**Comment by deanlee** [comment]
it's good enough. I'll create a sperate one.to merge it into https://github.com/commaai/openpilot/pull/26946#issuecomment-1383160270 and make it work with binview and historylog.
maybe it can be used to display signal values too:)

File: tools/cabana/binaryview.cc [hunk 1]
```diff
@@ -234,6 +237,7 @@ void BinaryItemDelegate::paint(QPainter *painter, const QStyleOptionViewItem &op
 
   if (index.column() == 8) {
     painter->setFont(hex_font);
+    painter->fillRect(option.rect, item->bg_color);
```
Comment (deanlee) [inline tools/cabana/binaryview.cc:240]: how about add some margins to fillrect ? e.g. 5?

Comment (pd0wm) [reply tools/cabana/binaryview.cc:240]: Tried that, but like it better without the gap between the colored squared.

**Comment by pd0wm** [comment]
@adeebshihadeh Is there anything you need to merge this PR? I have some other cool stuff ready that builds on top of this.

**Comment by deanlee** [comment]
I am also waiting for this PR to be merged to continue the next work.

File: tools/cabana/messageswidget.h [hunk 2]
```diff
@@ -41,3 +42,11 @@ class MessagesWidget : public QWidget {
   QString current_msg_id;
   MessageListModel *model;
 };
+
+class MessageBytesDelegate : public QStyledItemDelegate
+{
```
Comment (adeebshihadeh) [inline tools/cabana/messageswidget.h:47]:
```suggestion
class MessageBytesDelegate : public QStyledItemDelegate {
```

**Comment by adeebshihadeh** [comment]
> @adeebshihadeh Is there anything you need to merge this PR? I have some other cool stuff ready that builds on top of this.

Nope, just haven't checked out the backlog of cabana PRs in a bit. Either of you feel free to ping me if lack of merging is blocking your work!

**Pull request merged**